source/Plugins.cpp

```diff
@@ -50,6 +50,53 @@ namespace {
 
 
 
+// Checks if there are any dependencies of any kind.
+bool Plugin::PluginDependencies::IsEmpty() const
+{
+	return required.empty() && optional.empty() && conflicted.empty();
+}
+
+
+
+// Checks if there are any duplicate dependencies. E.g. the same dependency in both required and conflicted.
+bool Plugin::PluginDependencies::IsValid() const
+{
+	// We will check every dependency before returning to allow the
+	// plugin developer to see all errors and not just the first.
+	bool isValid = true;
+
+	// Required dependencies will already be valid due to sets not
+	// allowing duplicate values. Therefor we only need to check optional
+	// and conflicts.
+	for(const string &dependency : optional)
+	{
+		if(required.count(dependency))
+		{
+			Logger::LogError("Warning: Optional dependency with the name \"" + dependency
+				+ "\" was already found in required dependencies list.");
+		}
+	}
+	for(const string &dependency : conflicted)
+	{
+		if(required.count(dependency))
+		{
+			isValid = false;
+			Logger::LogError("Warning: Conflicts dependency with the name \"" + dependency
+				+ "\" was already found in required dependencies list.");
+		}
+		else if(optional.count(dependency))
```

Comment:
Seems like this could just be an `if` instead of an `else if`.
```suggestion
		if(optional.count(dependency))
```